examples/example_local_search_evolution_strategies.py

```diff
@@ -0,0 +1,174 @@
+
+import matplotlib.pyplot as plt
+import numpy as np
+import scipy.constants
+from docopt import docopt
+
+import cgp
+
+args = docopt(docopt_str)
+
+# %%
+# We first define the target function. Note that this function contains
+# numeric values which are initially not available as constants to the search.
+
+
+def f_target(x):
+    return np.e * x[:, 0] ** 2 + 1.0 + np.pi
+
+
+# %%
+# Then we define the objective function for the evolution. It consists
+# of an inner objective which accepts a NumPy-compatible function as
+# its first argument and returns the mean-squared error between the
+# expression represented by a given individual and the target function
+# evaluated on a set of random points. This inner objective is used by
+# the local search to determine appropriate values for Parameter node
+# and the actual objective function to update the fitness of the
+# individual.
+
+
+def inner_objective(f, seed):
+    """Return a loss for the numpy-compatible function f. Used for
+    calculating the fitness of each individual and for the local
+    search of numeric leaf values.
+
+    """
+
+    rng = np.random.RandomState(seed)
+    batch_size = 500
+    x = rng.uniform(-5, 5, size=(batch_size, 1))
+    y = f(x)
+    return -np.mean((f_target(x) - y[:, 0]) ** 2)
+
+
+def objective(individual, seed):
+    """Objective function of the regression task."""
+
+    if individual.fitness is not None:
+        return individual
+
+    f = individual.to_numpy()
+    individual.fitness = inner_objective(f, seed)
+
+    return individual
+
+
+# %%
+# Next, we define the parameters for the population, the genome of
+# individuals, the evolutionary algorithm, and the local search.
+
+
+population_params = {"n_parents": 1, "mutation_rate": 0.05, "seed": 818821}
+
+genome_params = {
+    "n_inputs": 1,
+    "n_outputs": 1,
+    "n_columns": 36,
+    "n_rows": 1,
+    "levels_back": None,
+    "primitives": (cgp.Add, cgp.Sub, cgp.Mul, cgp.Parameter),
+}
+
+ea_params = {"n_offsprings": 4, "tournament_size": 1, "n_processes": 1, "k_local_search": 2}
+
+evolve_params = {"max_generations": int(args["--max-generations"]), "min_fitness": 0.0}
+
+# restrict the number of steps in the local search; since parameter
+# values are propagated from parents to offsprings, parameter values
+# may be iteratively improved across generations despite the small
+# number of steps per generation
+local_search_params = {"max_generations": 5}
+
+# %%
+# We then create a Population instance and instantiate the local search
+# and evolutionary algorithm.
+
+pop = cgp.Population(**population_params, genome_params=genome_params)
+
+# define the function for local search; an instance of the
+# EvolutionStrategies can be called with an individual as an argument
+# for which the local search is performed
+local_search = cgp.local_search.EvolutionStrategies(
+    objective=functools.partial(inner_objective, seed=population_params["seed"]),
+    seed=population_params["seed"],
```

Comment:
isn't it a bit strange to use the same seed here as for the inner objective?

Reply:
maybe, not sure. what do you mean? ;)

Reply:
I mean, these are different random processes, so we should in principle use different seeds? Or one common RNG.
I don't think this has practical implications, but it seems a bit dirty to me to use the same seed in two places.

Reply:
yes, these are different random processes with different calls to the respective rngs. sure we can use different seeds if that is less confusing

Reply:
yes, please use a different seed. Just adding +1 would be sufficient, I guess.

Reply:
ok